mentor_classifier/mentor_classifier/arch/lr-transformer/predict.py

```diff
@@ -0,0 +1,121 @@
+#
+# This software is Copyright ©️ 2020 The University of Southern California. All Rights Reserved.
+# Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+#
+# The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+#
+import logging
+import os
+import random
+
+import joblib
+
+from mentor_classifier import (
+    QuestionClassifierPrediction,
+    QuestionClassiferPredictionResult,
+    mentor_model_path,
+    ARCH_LR_TRANSFORMER,
+)
+from mentor_classifier.api import create_user_question, OFF_TOPIC_THRESHOLD_DEFAULT
+from mentor_classifier.mentor import Mentor
+from mentor_classifier.utils import file_last_updated_at, sanitize_string
+
+
+class TransformersQuestionClassifierPrediction(QuestionClassifierPrediction):
+    def __init__(self, mentor, shared_root, data_path):
```

Comment:
always put the types in function args unless a strong reason to exclude, so here:

```python
def __init__(self, mentor: Union[str, Mentor], shared_root: str, data_path: str):
```